src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11898,9 +11912,7 @@ private static bool CanFollowCast(SyntaxKind kind)
                 case SyntaxKind.PercentToken:
                 case SyntaxKind.PlusPlusToken:
                 case SyntaxKind.MinusMinusToken:
-                // PROTOTYPE: We allow `(X)[...]` as a cast of a collection expression.  This could break existing
-                // code like (A.B.C)[0].
-                // case SyntaxKind.OpenBracketToken:
+                case SyntaxKind.OpenBracketToken:
```

Comment:
restored this to the original logic before i changed it in the feature branch.  always thinking of `[` as starting a literal after a `(...)` is likely too heavy a hammer.  But it might be good to bring to LDM.